lib/rspec/expectations/expectation_target.rb

```diff
@@ -118,9 +152,11 @@ def enforce_block_expectation(matcher)
       end
 
       def supports_block_expectations?(matcher)
-        matcher.supports_block_expectations?
-      rescue NoMethodError
-        false
+        if matcher.respond_to?(:supports_block_expectations?)
```

Comment:
This change is due to `have` matcher from `rspec-collection_matchers` that is sensitive to methods called on it, i.e. if we call `supports_value_expectations?` on `have(3).players`, it starts to think that it should match `supports_value_expectations?` association, not `players`.

The problem is with `supports_value_expectations?`, changing `supports_block_expectations?` for consistency.